src/smartpeak/include/SmartPeak/core/IProcessorDescription.h

```diff
@@ -0,0 +1,27 @@
+#pragma once
+#include <memory>
+
+namespace SmartPeak 
+{
+    struct IProcessorDescription 
+    {
+        /**
+            Get the processor struct ID
+        */
+        virtual int getID() const = 0;
+
+        /**
+            Get the processor struct name
+        */
+        virtual std::string getName() const = 0;
+
+        /**
+            Get the processor struct description
+        */
+        virtual std::string getDescription() const = 0;
+
+        virtual ~IProcessorDescription() = default;
+    };
+
+    typedef std::shared_ptr<IProcessorDescription> IProcessorDescriptionPtr;
```

Comment:
This declaration can be usefull, but I would prefer a team agreement and to have a check on all code to use this pattern in a dedicated story to have consistent code. my concern is to have this pattern use only on some parts of code and not on others. to me explicit shared_ptr is nice because you can also have unique_ptr the declaration my hide this.

Reply:
Agree with your concerns, its a good practice to be consistent and keep same `coding conventions` everywhere in the project. We can have some discussion later during sprint review/retro.